imap_processing/swe/tests/test_swe_science.py

```diff
@@ -20,20 +43,126 @@ def test_uncompress_algorithm():
-    """
-    # First ESA in test data is 1.
-    data = swe_science(decom_test_data)
-    first_esa_steps = data["ESA_STEPS"][0]
-    assert first_esa_steps == 1
+    """Test general shape of return dataset from swe_science."""
+    # Get unpacked science data
+    processed_data = swe_science(decom_test_data)
 
     # science data should have this shape, 15x12x7.
-    science_data = data["SCIENCE_DATA"][0]
+    science_data = processed_data["SCIENCE_DATA"].data[0]
     assert science_data.shape == (15, 12, 7)
 
-    # Test data has 23 packets, therefore, SPIN_PHASE should have this length.
-    spin_phase = data["SPIN_PHASE"]
-    assert len(spin_phase) == 23
+    # Test data has n packets, therefore, SPIN_PHASE should have that same length.
+    spin_phase = processed_data["SPIN_PHASE"]
+    expected_length = 4
+    assert len(spin_phase) == expected_length
+
+
+def test_decompress_counts(decom_test_data):
+    """Test decompress counts."""
+    test_data_path = f"{imap_module_directory}/swe/tests/decompressed"
+    filepaths = [
+        "20230927173253_1st_quarter_decompressed.csv",
+        "20230927173308_2nd_quarter_decompressed.csv",
+        "20230927173323_3rd_quarter_decompressed.csv",
+        "20230927173238_4th_quarter_decompressed.csv",
+    ]
+    uncompressed_data = swe_science(decom_test_data)
+
+    for index in range(len(filepaths)):
+        instrument_uncompressed_counts = pd.read_csv(
+            f"{test_data_path}/{filepaths[index]}", index_col="Index"
+        )
+
+        assert (
+            uncompressed_data["QUARTER_CYCLE"].data[index]
+            == decom_test_data[index].data["QUARTER_CYCLE"].raw_value
+        )
+        sdc_uncompressed_counts = (
+            uncompressed_data["SCIENCE_DATA"].data[index].reshape(180, 7)
+        )
+
+        cem_1_decompresed_counts = instrument_uncompressed_counts["CEM 1"].values
+        cem_2_decompresed_counts = instrument_uncompressed_counts["CEM 2"].values
+        cem_3_decompresed_counts = instrument_uncompressed_counts["CEM 3"].values
+        cem_4_decompresed_counts = instrument_uncompressed_counts["CEM 4"].values
+        cem_5_decompresed_counts = instrument_uncompressed_counts["CEM 5"].values
+        cem_6_decompresed_counts = instrument_uncompressed_counts["CEM 6"].values
+        cem_7_decompresed_counts = instrument_uncompressed_counts["CEM 7"].values
+
+        # compare value of each CEM
+        assert np.all(sdc_uncompressed_counts[:, 0] == cem_1_decompresed_counts)
+        assert np.all(sdc_uncompressed_counts[:, 1] == cem_2_decompresed_counts)
+        assert np.all(sdc_uncompressed_counts[:, 2] == cem_3_decompresed_counts)
+        assert np.all(sdc_uncompressed_counts[:, 3] == cem_4_decompresed_counts)
+        assert np.all(sdc_uncompressed_counts[:, 4] == cem_5_decompresed_counts)
+        assert np.all(sdc_uncompressed_counts[:, 5] == cem_6_decompresed_counts)
+        assert np.all(sdc_uncompressed_counts[:, 6] == cem_7_decompresed_counts)
```

Comment:
Could you instead do something like this?

```python
for i in range(7):
    cem_decompressed_counts = instrument_uncompressed_counts[f"CEM {i+1}"].values
    assert np.all(sdc_uncompressed_counts[:, i] == cem_decompressed_counts
```

Reply:
for sure! will change that